### ISSUE TITLE
accessibleText computed for div with CSS background image is wrong

### ISSUE THREAD

**Issue by iamrafan** [dequelabs/axe-core#1596]
Demo: https://codepen.io/anon/pen/LoeXoV

Expectation: accessibleText computed for ```div``` with id ```"parent-element"``` is ""

Actual: accessibleText computed for ```div``` with id ```"parent-element"``` is "Test label for nested element"

Motivation: According to [accessible name computation algorithm](https://www.w3.org/TR/accname-1.1/#terminology) div should not be getting the name from content as it is not one of the [roles that get name from content](https://www.w3.org/TR/wai-aria/#namefromcontent)

<pre><code>
axe-core version: 3.2.2
</code></pre>

**Comment by straker** [comment]
Thanks for reporting the issue. Without digging too much into this, I'm wondering if it's related to #1398

**Comment by iamrafan** [comment]
@straker I too think so. Both these elements do not have the [roles that allow name from content](https://www.w3.org/TR/wai-aria/#namefromcontent)

**Comment by AutoSponge** [comment]
The accName calc should skip "genericContainer" elements.

```js
// has no explicit role
// doesn't map to an element with an implicit role
// has no textNode children with content
Array.from(el.childNodes).filter(n => n.nodeType === Node.TEXT_NODE && n.textContent.trim().length).length
// the above is needed because of how DOM with whitespace is parsed into textNodes
```

**Comment by WilcoFiers** [comment]
Hey guys, sorry to throw this back, but this behaviour is intentional. Axe-core's `accessibleText` function is not intended as an accurate implementation of the accessible name computation algorithm. If you're looking to use it that way, you need to pass it `{ strict: true }`.

@iamrafan if this came up because of a false positive, I'd like to know that. But if not, this isn't something I think we should fix.

We should revisit this after we've sorted out https://github.com/dequelabs/axe-core/issues/1611. I think this is one of the bigger reasons for us having this divergent behaviours.

**Comment by dylanb** [comment]
@iamrafan we have reported an issue to the working group because the accessible name computation algorithm has a bug that would result in empty accessible names in situations where this is not wanted and also we have documented that not a single browser-SR combination actually implements either the entire algorithm as documented nor the algorithm as intended. There are differences between them all. axe-core implements a "common denominator" approach that is consistent with the WCAG accessibility supported requirement.
